addons/project/models/project_task.py

```diff
@@ -1087,6 +1088,38 @@ def update_date_end(self, stage_id):
             return {'date_end': fields.Datetime.now()}
         return {'date_end': False}
 
+    def search_on_comodel(self, domain, field, comodel, order=None, additional_domain=None):
+        filtered_domain = filter_domain_leaf(domain, lambda field_to_check: field_to_check in [
+            field,
+            f"{field}.id",
+            f"{field}.name",
+        ], {
+            field: "name",
+            f"{field}.id": "id",
+            f"{field}.name": "name",
+        })
+        if not filtered_domain:
+            return False
+        if additional_domain:
+            filtered_domain = expression.AND([filtered_domain, additional_domain])
+        return self.env[comodel].search(self.change_operator(filtered_domain), order=order)
+
+    def change_operator(self, domain):
```

Comment:
Same comment here. 🙂

Reply:
I am wondering if we should not do it inside `_search_on_comodel` since it is really specific to that method. In another words, I am not sure we will really use it somewhere else. 🙂

Reply:
Yes, I just wanted to split the function to be more clear. But I guess I can declare de function `_change_operator` in the other function ?

Reply:
Yep good idea, it is just because it is really specific to `_search_on_comodel` and so I am not sure we will use that function in another use case. 🙂 

If one day yes then indeed that method can be defined in the model. 🙂